doc/decisions/0007-feature-toggles-development.md

```diff
@@ -0,0 +1,92 @@
+# 0007. Feature toggles development
+
+| Date | Tags |
+|---|---|
+| 2018-05-25 | development |
+
+
+## Status
+
+Proposed
+
+## Context
+
+This change describes a methodology of development which we believe will help
+us increase the confidence and speed in the development process.
+
+### What problem are we trying to solve
+
+Currently the time between a PR is sent and the code gets merged is between 1 day
+to in some case 4 to 5 days.
+
+This has some disadvantages:
+
+1) This workflows incentivise developers to make large PRs, as building 
+incrementally but having to wait 2 to 3 days for the code to be merged at each
+step is not desirable and results in loss of context.
+
+2) If a bug is introduced it is harder to fix, as the longer it takes to integrate code 
+the more likely the developer has lost familiarity with the code.
+
+3) Constant rebases introduce an overhead for the developer and can introduce
+subtle bugs, as rebased code is not thoroughly reviewed.
```

Comment:
Strongly agree. In fact, we've had cases that even automatic rebases without any apparent conflicts produced code that doesn't even compile.

Reply:
Automatic rebase should be disabled imho, it's just dangerous.